src/Cookie.php

```diff
@@ -492,7 +492,22 @@ public static function parse($cookie_header, $name = '', $reference_time = null)
 	 * @param int|null                         $time    Reference time for expiration calculation
 	 * @return array
 	 */
-	public static function parse_from_headers(Headers $headers, Iri $origin = null, $time = null) {
+	public static function parse_from_headers(Headers $headers, $origin, $time = null) {
+		// Mimic a \WpOrg\Requests\Iri|null type-check for $origin.
+		// Change to a type declaration when PHP >= 7.1 is the base requirement.
+		if ($origin !== null && !$origin instanceof \WpOrg\Requests\Iri) {
+			if (\PHP_VERSION_ID >= 70000) {
+				throw new \TypeError(
+					'WpOrg\Requests\Cookie::parse_from_headers(): Argument #2 ($origin) must be of type WpOrg\Requests\Iri, ' . gettype($origin) . ' given'
+				);
+			}
+			trigger_error(
+				'Argument 2 passed to WpOrg\Requests\Cookie::parse_from_headers() must be an instance of WpOrg\Requests\Iri, ' . gettype($origin) . ' given',
+				E_USER_ERROR
+			);
+			exit();
+		}
```

Comment:
I think this can be simplified by using the `WpOrg\Requests\Exception\InvalidArgument` exception, which we're also using everywhere else we're doing input validation in-function in Requests:

```php
		if (!empty($origin) && !($origin instanceof Iri)) {
			throw InvalidArgument::create(2, '$origin', Iri::class, gettype($origin));
		}
```

Also take note of the parentheses around the `instanceof`. These are necessary as otherwise the precedence order will cause this code to be read as `(!$origin) instanceof Iri`, which is effectively `true|false instanceof Iri`, which would always be `false`.

The function will now also need to get a `@throws` tag.

As a last point, I wonder if this check should be moved to below the `$cookie_headers` check ? as that can function just fine without the `$origin` being validated.

Reply:
Oh and before I forget: this change will also need a test to cover it.